llvm/utils/TableGen/Common/CodeGenDAGPatterns.cpp

```diff
@@ -1600,18 +1601,13 @@ static TreePatternNode &getOperandNum(unsigned OpNo, TreePatternNode &N,
     return N;
   }
 
-  OpNo -= NumResults;
-
-  if (OpNo >= N.getNumChildren()) {
-    std::string S;
-    raw_string_ostream OS(S);
-    OS << "Invalid operand number in type constraint " << (OpNo + NumResults)
-       << " ";
-    N.print(OS);
-    PrintFatalError(S);
+  if (OpNo - NumResults >= N.getNumChildren()) {
+    PrintFatalError([&N, OpNo](raw_ostream &OS) {
+      OS << formatv("Invalid operand number {0} in type constraint ", OpNo);
```

Comment:
Doesn't formatv create a buffer to hold the full string before printing? Would `OS << "Invalid operand number " << OpNo << "in type constraint" be better?

Reply:
AFAIK it does not. It splits the format string into fragments and prints them one-by-one. Though in this case, may be what you suggested is better (in the sense what we are printing is relatively simple and does not warrant use of formatv). I'll change it.

Reply:
Done.